src/controller/python/chip/clusters/Attribute.py

```diff
@@ -723,9 +724,12 @@ def handleEventData(self, header: EventHeader, path: EventPath, data: bytes, sta
                 Header=header, Data=eventValue, Status=chip.interaction_model.Status(status))
             self._events.append(eventResult)
 
-            if (self._subscription_handler is not None):
-                self._subscription_handler.OnEventChangeCb(
-                    eventResult, self._subscription_handler)
+            if self._subscription_handler is not None:
+                callback = self._subscription_handler.OnEventChangeCb
+                if inspect.iscoroutinefunction(callback):
+                    asyncio.run_coroutine_threadsafe(callback(), self._event_loop)
```

Comment:
shouldn't the arguments be the same regardless if it will be run in the event loop or not?

Reply:
Right, good catch!